Studio comments off

picklesrus opened the pull request:
Adds in comment status component to show when studio comments are off. 
![image](https://user-images.githubusercontent.com/15675877/121595465-4de1e400-ca0c-11eb-9ba4-66d2269362e4.png)

picklesrus commented on src/views/studio/studio-comments.jsx:
```diff
@@ -74,54 +75,67 @@ const StudioComments = ({
                 <h2><FormattedMessage id="studio.commentsHeader" /></h2>
                 {canEditCommentsAllowed && <StudioCommentsAllowed />}
             </div>
```
This part of the diff is much easier to read if you turn whitespace changes off.

paulkaplan commented on src/redux/session.js:
```diff
@@ -142,6 +142,8 @@ module.exports.selectIsSocial = state => get(state, ['session', 'session', 'perm
 module.exports.selectIsEducator = state => get(state, ['session', 'session', 'permissions', 'educator'], false);
 module.exports.selectProjectCommentsGloballyEnabled = state =>
     get(state, ['session', 'session', 'flags', 'project_comments_enabled'], false);
+module.exports.selectStudioCommentsGloballyEnabled = state =>
```
@picklesrus I think that this should probably be true by default right until we hear from the session? I've been trying to make sure the path to trying to load stuff is as parallel as possible, and this change would prevent loading the comments list until the user session has been fetched, which isn't necessary.

paulkaplan commented on src/views/studio/studio-comments.jsx:
```diff
@@ -74,54 +75,67 @@ const StudioComments = ({
-                {shouldShowCommentComposer ?
-                    (commentsAllowed ?
-                        <ComposeComment
+            {!hasFetchedSession || studioCommentsGloballyEnabled ?
```
Why is the `!hasFetchedSession` here blocking the comments from rendering? possibly related to my other comment about not blocking comment loading with session request

paulkaplan reviewed (changes requested):
A couple of questions/requests for changes. I'm happy to talk about the questions IRL.

picklesrus commented:
This isn't finished (e.g.tests would need to be cleaned up), but I'm curious if you like this approach better

paulkaplan commented:
I approve of the direction. I like the unit test on studio-comments and probably would want a unit test on the permission selectors as well.

picklesrus commented:
Simplified the tests in studio-comments and moved them to studio-permissions. Glad they were there because they found a bug in the refactor. Fixed that and resolved some merge conflicts. Ready for another review!

paulkaplan reviewed (approved):
LGTM

paulkaplan commented:
Sorry about the continuing merge conflicts, there is a lot going on with studio comments right now. Should be settled now for at least a few days, but definitely recommend merging asap

picklesrus commented:
I added some test cases to handle the single comment mode. If they pass on Circle, I'm merging. :)
